Introduce isDeprecated to CallableParametersAcceptor

kamil-zacek opened the pull request.

ondrejmirtes committed:
Issue bot - let all comments about improved `ArrayType::describe()` through

Chemaclass committed:
test: use bashunit -a exit_code to check for errors

and defer the execution call inside bashunit

staabm committed:
Remove $isFinal dead-code in PhpFunctionReflection

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

ondrejmirtes committed:
IntersectionType - always describe list as list

ondrejmirtes committed:
TypeNodeResolver - more specific array key type for lists

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

ondrejmirtes committed:
Merge remote-tracking branch 'origin/1.12.x' into 2.0.x

ondrejmirtes committed:
[BCB] Remove `ConstantArrayType::slice()`

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

herndlm committed:
Get rid of unnecessary `instanceof self` in `ConstantArrayType`

herndlm committed:
Get rid of unnecessary `instanceof self` in `ConstantArrayType`

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

ondrejmirtes committed:
Fix stubs

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

ondrejmirtes committed:
Merge remote-tracking branch 'origin/1.12.x' into 2.0.x

ondrejmirtes committed:
Merge remote-tracking branch 'origin/1.12.x' into 2.0.x

ondrejmirtes committed:
[BCB] Changed `TemplateType::isValidVariance()` return type to IsSuperTypeOfResult

ondrejmirtes committed:
[BCB] Remove `Type::isSuperTypeOfWithReason()`, `Type::isSuperTypeOf()` return type changed to IsSuperTypeOfResult

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

ondrejmirtes committed:
Do not comment on PRs on 2.0.x anymore

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

ondrejmirtes committed:
Update changelog-2.0.md

ondrejmirtes committed:
Fix

Slamdunk committed:
UPGRADING.md: typo

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

ondrejmirtes committed:
Revert "Get rid of unnecessary `instanceof self` in `ConstantArrayType`"

This reverts commit f818ac594f1ac9e3c072aa13888921fffa47b29c.

ondrejmirtes committed:
Merge remote-tracking branch 'origin/1.12.x' into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

phpstan-bot committed:
Update PHP 8 stubs

ondrejmirtes committed:
Fix

ondrejmirtes committed:
Fix generate-function-metadata.php script

staabm committed:
Remove inefficient caching from `PhpMethodReflection` and `PhpFunctionReflection::isVariadic()`

Co-authored-by: Ondrej Mirtes <ondrej@mirtes.cz>

ondrejmirtes committed:
Cleanup - visitor already registered

ondrejmirtes committed:
Fixed optimization

ondrejmirtes committed:
Clean file cache from unused items

ondrejmirtes committed:
This was also used format

ondrejmirtes committed:
Update cache deletion logic

ondrejmirtes committed:
Journal for used generated containers

ondrejmirtes committed:
Merge remote-tracking branch 'origin/1.12.x' into 2.0.x

ondrejmirtes committed:
Update nikic/php-parser

ondrejmirtes committed:
Revert "Fix generate-function-metadata.php script"

This reverts commit 14a3b36d72f149fda2837cf11cc899db5aa87a7a.

phpstan-bot committed:
Update PHP 8 stubs

ondrejmirtes committed:
Update PhpStorm stubs

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

ondrejmirtes committed:
Remove old containers from the journal

ondrejmirtes committed:
Fix CS

jiripudil committed:
Merge branch '1.12.x' into 2.0.x

ondrejmirtes committed:
Merge remote-tracking branch 'origin/1.12.x' into 2.0.x

ondrejmirtes committed:
Move template-default.php to nsrt

ondrejmirtes committed:
Test template-default.php only on PHP 8+

staabm committed:
Added regression test

ondrejmirtes committed:
Update simple-downgrader

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

ondrejmirtes committed:
Merge remote-tracking branch 'origin/1.12.x' into 2.0.x

ondrejmirtes committed:
Merge remote-tracking branch 'origin/1.12.x' into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

ondrejmirtes committed:
Revert "Simulate level 10 in issue bot on 1.12.x"

This reverts commit def9f5abccb4b2e761dc7886f444e56aae25d97a.

ondrejmirtes committed:
Refactor IntersectionType::describe()

ondrejmirtes committed:
Fix test

ondrejmirtes committed:
ArrayType::describe - explicit mixed should be stated explicitly

ondrejmirtes committed:
Revert "Issue bot - let all comments about improved `ArrayType::describe()` through"

This reverts commit 4007e653b9a9fe9db7d8c4b9d977ca4175786ff1.

ondrejmirtes committed:
Merge remote-tracking branch 'origin/1.12.x' into 2.0.x

ondrejmirtes committed:
Fix build

ondrejmirtes committed:
StubParser

ondrejmirtes committed:
Fix tests

ondrejmirtes committed:
Fix tests

ondrejmirtes committed:
Use StubParser more

ondrejmirtes committed:
Fix tests

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

ondrejmirtes committed:
Fixed types

ondrejmirtes committed:
Separate FileTypeMapper for StubPhpDocProvider

ondrejmirtes committed:
Fixed tests

ondrejmirtes committed:
Fixed tests

ondrejmirtes committed:
Do not run integration tests on PHAR for now

ondrejmirtes committed:
Merge branch '1.12.x' into 2.0.x

ondrejmirtes committed:
Merge branch '1.12.x' into 2.0.x

ondrejmirtes committed:
Upgrading notes about 1st party extensions

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

ondrejmirtes committed:
Update PhpStorm stubs

staabm committed:
Remove dead code

phpstan-bot committed:
Update PHP 8 stubs

ondrejmirtes committed:
Merge remote-tracking branch 'origin/1.12.x' into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

ondrejmirtes committed:
Update PhpStorm stubs

ondrejmirtes committed:
Fix build

ondrejmirtes committed:
Merge remote-tracking branch 'origin/1.12.x' into 2.0.x

ondrejmirtes committed:
Merge remote-tracking branch 'origin/1.12.x' into 2.0.x

ondrejmirtes committed:
Fix CS

ondrejmirtes committed:
Merge remote-tracking branch 'origin/1.12.x' into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

ondrejmirtes committed:
Update PhpStorm stubs

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

phpstan-bot committed:
Update PHP 8 stubs

u01jmg3 committed:
Support returning an array or a string in `count_chars()`

staabm committed:
Remove dead code in ConstantConditionRuleHelper

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

ondrejmirtes committed:
Update PhpStorm stubs

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

ondrejmirtes committed:
Merge remote-tracking branch 'origin/1.12.x' into 2.0.x

janedbal committed:
xdebug_get_function_stack: fix signature

ondrejmirtes committed:
Update changelog

staabm committed:
Utilize PHP version constraint from composer.json to narrow `PHP_*` constants

ondrejmirtes committed:
Merge remote-tracking branch 'origin/1.12.x' into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

ondrejmirtes committed:
Merge remote-tracking branch 'origin/1.12.x' into 2.0.x

ondrejmirtes committed:
Reorganize new rules in changelog

ondrejmirtes committed:
Update phpdoc-parser

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

ondrejmirtes committed:
Fix after merge

ruudk committed:
Use named argument in error for variadic types

When a type is variadic, and multiple named arguments are used, it's often hard
to tell for which argument the error is coming.

Since we have the named argument, we could use it in this situation.

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

ondrejmirtes committed:
Fix after merge

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

Chemaclass committed:
Upgrade bashunit:0.18.0 for e2e tests

kamil-zacek committed:
Introduce isDeprecated to CallableParametersAcceptor

ondrejmirtes commented on src/Type/ClosureType.php:
```diff
@@ -143,6 +143,11 @@ public function isPure(): TrinaryLogic
 
+	public function isDeprecated(): TrinaryLogic
+	{
+		return TrinaryLogic::createNo();
```
This is not the case. First-class callables like `Foo::method(...)` are resolved to ClosureType so the deprecated-ness should be carried over.

kamil-zacek replied:
I am not sure how to deal with it. Do you think the class ClosureType itself should have a member "isDeprecated" which will be in the constructor and resolved during the creation of instance of ClosureType itself?

ondrejmirtes reviewed (changes requested):
Also please review all `new CallableType` to make sure we haven't missed a place where it should be added.

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

ondrejmirtes committed:
Update UPGRADING.md

ondrejmirtes committed:
Update changelog one last time

ondrejmirtes committed:
Update PHPStan extensions

ondrejmirtes committed:
Remove changelog

ondrejmirtes committed:
Update issue-bot

ondrejmirtes committed:
Upgrading guide will only be in phpstan/phpstan

ondrejmirtes committed:
Revert "Upgrading guide will only be in phpstan/phpstan"

This reverts commit 6517446a99d70c75015e6fc2990adba63a662913.

canvural committed:
fix: check for existence of second arg in CountCharsFunctionDynamicReturnTypeExtension

ondrejmirtes committed:
Fix resolving tentative return type

ondrejmirtes committed:
Merge remote-tracking branch 'origin/1.12.x' into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

ondrejmirtes committed:
Fix test

ondrejmirtes committed:
Update PhpStorm stubs

phpstan-bot committed:
Update PHP 8 stubs

ondrejmirtes committed:
Run integration tests again

ondrejmirtes committed:
Merge remote-tracking branch 'origin/1.12.x' into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

phpstan-bot committed:
Update PHP 8 stubs

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

staabm committed:
Refactor ComposerPhpVersionFactory, ConstantResolver

staabm committed:
Utilize phpVersion.min+max in VersionCompareFunctionDynamicReturnTypeExtension

staabm committed:
More details php version information in diagnose

ondrejmirtes committed:
Merge remote-tracking branch 'origin/1.12.x' into 2.0.x

ondrejmirtes committed:
Merge remote-tracking branch 'origin/1.12.x' into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

ondrejmirtes committed:
Fix after merge

ondrejmirtes committed:
Merge remote-tracking branch 'origin/1.12.x' into 2.0.x

staabm committed:
FunctionCallParametersCheck: Add native parameter type

sreichel committed:
3rd parameter of htmlentities|htmlspecialchars allows null

ondrejmirtes committed:
Update PhpStorm stubs

ondrejmirtes committed:
Merge remote-tracking branch 'origin/1.12.x' into 2.0.x

ondrejmirtes committed:
Fix test

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

xPaw committed:
Update curl_setopt string values and allow nullable

claudepache committed:
bccomp: more precise return type

ondrejmirtes committed:
Update BetterReflection

ondrejmirtes committed:
Merge remote-tracking branch 'origin/1.12.x' into 2.0.x

ondrejmirtes committed:
Merge branch '1.12.x' into 2.0.x

ondrejmirtes committed:
Regenerate baseline

ondrejmirtes committed:
Merge remote-tracking branch 'origin/1.12.x' into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

janedbal committed:
Fix subtracting enums inside `in_array`

Co-authored-by: Ondrej Mirtes <ondrej@mirtes.cz>

janedbal committed:
TooWidePropertyTypeRule: dont skip even always-read properties

VincentLanglet committed:
Refactor TryRemove/Accepts for DateTime|DateTimeImmutable and Exception|Error

staabm committed:
Implement Scope->getPhpVersion()

ondrejmirtes committed:
Revert "TooWidePropertyTypeRule: dont skip even always-read properties"

This reverts commit adcabb3078603f46c811c7dc460174c9df551310.

schlndh committed:
Bleeding edge - check that values passed to array_sum/product are castable to number

staabm committed:
`Scope::getPhpVersion()` allows array via phpVersion min+max config

staabm committed:
non-capturing catch support-detection is scope php-version dependent

sayuprc committed:
Last value was not recognized when passing an associative array as an argument

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

ondrejmirtes committed:
Update PhpStorm stubs

ondrejmirtes committed:
Update BetterReflection

ondrejmirtes committed:
Update BetterReflection

phpstan-bot committed:
Update PHP 8 stubs

ondrejmirtes committed:
PropertyHookType should not be prefixed in BetterReflection

ondrejmirtes committed:
Update BetterReflection

samsonasik committed:
Remove shortArraySyntax definiton on Printer::construct()

It seems alreayd short array syntax by default

ondrejmirtes committed:
Un-finalize Printer

See https://github.com/rectorphp/rector-src/pull/6517#issuecomment-2503502706

ondrejmirtes committed:
Revert "Un-finalize Printer"

This reverts commit 1abeeb77771fc3fece5c7196f75dfc1478840515.

ondrejmirtes committed:
Update BetterReflection

ondrejmirtes committed:
Fix CS

ondrejmirtes committed:
Update PhpStorm stubs

ondrejmirtes committed:
Merge remote-tracking branch 'origin/1.12.x' into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

phpstan-bot committed:
Update PHP 8 stubs

ondrejmirtes committed:
Update BetterReflection

ondrejmirtes committed:
Update BetterReflection

jakubtobiasz committed:
Sanity checks around hooked properties in interfaces and classes

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

schlndh committed:
skip param castable to X on non-arrays

Fixes bug 12146

ondrejmirtes committed:
Merge remote-tracking branch 'origin/1.12.x' into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

ondrejmirtes committed:
Merge remote-tracking branch 'origin/1.12.x' into 2.0.x

phpstan-bot committed:
Update PHP 8 stubs

xPaw committed:
Remove incorrect CURLOPT_ACCEPT_ENCODING alias

ondrejmirtes committed:
Update PhpStorm stubs

jack-worman committed:
Fix-GH-12021

DemvSteinbrink committed:
Workaround bug in slevomat/coding-standard TypeNameMatchesFileName

For each root namespace, the slevomat rule considers the left-most
match of the given directory in the absolute path of the file. That is,
for /home/user/src/phpstan-src/ the root namespace PHPStan is not
assigned to /home/user/src/phpstan-src/src, but to /home/user/src, which
is obviously wrong.

The bug is known as slevomat/coding-standard#1249 for a long time, but
yet to be fixed. To avoid issues for developers of PHPStan, we can set
a basepath of "." in the PHP CodeSniffer config, which causes paths to
be evaluated relative to the current directory, avoiding
false-positives in the path leading up to the phpstan-src directory.

janedbal committed:
Fix `fgetcsv` return type; never returns null

ondrejmirtes committed:
Merge remote-tracking branch 'origin/1.12.x' into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

ondrejmirtes committed:
Fix after merge

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

ondrejmirtes committed:
Merge remote-tracking branch 'origin/1.12.x' into 2.0.x

ondrejmirtes committed:
Merge remote-tracking branch 'origin/1.12.x' into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

vindic committed:
Fix apcu_cache_info and apcu_sma_info signatures

ondrejmirtes committed:
Make these benevolent

ondrejmirtes committed:
Update BetterReflection

staabm committed:
Added `strictRulesInstalled` parameter

AJenbo committed:
Remove incorrect doc leftover from 1.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

ondrejmirtes committed:
Update PhpStorm stubs

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

ondrejmirtes committed:
Merge remote-tracking branch 'origin/1.12.x' into 2.0.x

staabm committed:
Readonly classes cannot be combined with `#[AllowDynamicProperties]` + check trait attributes

staabm committed:
Named argument detection is scope-PHP version dependent

phpstan-bot committed:
Merge branch refs/heads/1.12.x into 2.0.x

ondrejmirtes committed:
Fix test

ondrejmirtes committed:
Attributes rules use `In*Node` virtual nodes for more precise Scope

staabm committed:
Scope: use `scope->getConstant` instead

janedbal committed:
Bleeding edge - UnusedFunctionParametersCheck: report precise line

ondrejmirtes committed:
Update BetterReflection

ondrejmirtes committed:
Prepare for 2.1.x-dev

staabm commented:
I guess rhis was rebased against 2.1.x but the PR target needs still to be changed to 2.1.x

kamil-zacek commented:
> I guess rhis was rebased against 2.1.x but the PR target needs still to be changed to 2.1.x

Yes, I know it. I will close and reopen PR when I receive reply to my previous question.
